WalletWasabi/WabiSabi/Client/CoinJoinManager.cs

```diff
@@ -50,7 +50,8 @@ public CoinJoinManager(IWalletProvider walletProvider, RoundStateUpdater roundSt
 	/// </summary>
 	private ConcurrentDictionary<string, byte> WalletsInSendWorkflow { get; } = new();
 
-	public CoinJoinClientState HighestCoinJoinClientState { get; private set; }
+	public CoinJoinClientState HighestCoinJoinClientState => CoinJoinClientStates.Values.MaxBy(s => (int)s);
+	public ImmutableDictionary<string, CoinJoinClientState> CoinJoinClientStates { get; private set; } = ImmutableDictionary<string, CoinJoinClientState>.Empty;
```

Comment:
If it's for private use only is there a reason not to use mutable collections?

Reply:
I will use it in the next PR. Until that it will be private.

Reply:
Aha, and the next usage question is do you really need a thread safe dictionary here? Checking it once more, it doesn't look like you need previous elements or even the whole dictionary instance, so `ReadOnlyDictionary<,>` might be a better choice. Look [at this benchmark](https://gist.github.com/nycdotnet/008ed7a417b6dca119323e25eb7aabe4).

With .NET 8 another dictionary type will be shipped, `FrozenDictionary<,>. It's a hybrid dictionary type which is an array for a small element number and a normal dictionary otherwise, tuned for read operations.

Reply:
Makes sense, I will open an issue for that then.